src/components/CustomAppDetails/index.js

```diff
@@ -0,0 +1,291 @@
+            <Button secondary onClick={handleUninstall}>
+                {i18n.t('Uninstall')}
+            </Button>
+        </div>
+    )
+}
+
+ManageInstalledVersion.propTypes = {
+    reloadPage: PropTypes.func.isRequired,
+    versions: PropTypes.array.isRequired,
+    installedApp: PropTypes.object,
+}
+
+const Metadata = ({ installedVersion, versions }) => {
+    const relativeTime = datetime => moment(datetime).fromNow()
+    versions = versions.sort((a, b) => a.created - b.created)
+    const latestVersion = getLatestVersion(versions)
+
+    return (
+        <ul className={styles.metadataList}>
+            <li className={styles.metadataItem}>
+                {i18n.t('Version {{version}}', {
+                    version: installedVersion || latestVersion.version,
+                })}
+            </li>
+            <li className={styles.metadataItem}>
+                {i18n.t('Last updated {{relativeTime}}', {
+                    relativeTime: relativeTime(
+                        versions[versions.length - 1].created
+                    ),
+                })}
+            </li>
+            <li className={styles.metadataItem}>
+                {i18n.t('First published {{relativeTime}}', {
+                    relativeTime: relativeTime(versions[0].created),
+                })}
+            </li>
+        </ul>
+    )
+}
+
+Metadata.propTypes = {
+    versions: PropTypes.array.isRequired,
+    installedVersion: PropTypes.string,
+}
+
+const Screenshots = ({ screenshots }) => {
+    const [currentScreenshotIndex, setCurrentScreenshotIndex] = useState(0)
+    const currentScreenshot = screenshots[currentScreenshotIndex]
+
+    return (
+        <div className={styles.screenshots}>
+            <div>
+                <img
+                    className={styles.currentScreenshot}
+                    src={currentScreenshot}
+                />
+            </div>
+            <div>
+                {screenshots.map((screenshot, index) => (
+                    <button
+                        key={index}
+                        onClick={() => setCurrentScreenshotIndex(index)}
+                        className={
+                            styles.otherScreenshot +
+                            ' ' +
+                            (index === currentScreenshotIndex
+                                ? styles.otherScreenshotCurrent
+                                : '')
+                        }
+                    >
+                        <img src={screenshot} />
+                    </button>
+                ))}
+            </div>
+        </div>
+    )
+}
+
+Screenshots.propTypes = {
+    screenshots: PropTypes.array.isRequired,
+}
+
+const CustomAppDetails = ({ match }) => {
+    const query = {
+        app: {
+            resource: `appHub/v1/apps/${match.params.appHubId}`,
+        },
+        installedApps: {
+            resource: 'apps',
+        },
+    }
+    const { loading, error, data, refetch } = useDataQuery(query)
```

Comment:
We support dynamic query `id`, `params`, and `data` properties with a static query using query variables, so this can be rewritten like so:

```js
const query = {
    app: {
        resource: `appHub/v1/apps`,
        id: ({ appHubId }) => appHubId,
    },
    installedApps: {
        resource: 'apps',
    },
}
const CustomAppDetails = ({ match }) => {
    const { loading, error, data, refetch } = useDataQuery(query, { variables: { appHubId: match.params.appHubId } })
```